tests/integ/modin/index/test_reindex.py

```diff
@@ -0,0 +1,269 @@
+def test_reindex_index_basic_add_elements():
+    native_idx = native_pd.Index(list("ABC"))
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex(list("CABDEF"))
+    snow_result = snow_idx.reindex(list("CABDEF"))
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_basic_remove_elements():
+    native_idx = native_pd.Index(list("ABC"))
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex(list("CA"))
+    snow_result = snow_idx.reindex(list("CA"))
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_basic_add_remove_elements():
+    native_idx = native_pd.Index(list("ABC"))
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex(list("CADEFG"))
+    snow_result = snow_idx.reindex(list("CADEFG"))
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("method", ["bfill", "backfill", "pad", "ffill"])
+@pytest.mark.parametrize("limit", [None, 1, 2, 100])
+def test_reindex_index_fill_method(method, limit):
+    native_idx = native_pd.Index(list("ACE"))
+    snow_idx = pd.Index(native_idx)
+    native_result = native_idx.reindex(list("ABCDEFGH"), method=method, limit=limit)
+    snow_result = snow_idx.reindex(list("ABCDEFGH"), method=method, limit=limit)
+    assert_reindex_result_equal(snow_result, native_result)
+
+
+@pytest.mark.parametrize("method", ["bfill", "backfill", "pad", "ffill"])
+@pytest.mark.parametrize("limit", [None, 1, 2, 100])
+def test_reindex_index_fill_method_non_monotonic_new_index(method, limit):
+    # The negative is because pandas does not support `limit` parameter
+    # if the index and target are not both monotonic, while we do.
+    native_idx = native_pd.Index(list("ABC"))
+    snow_idx = pd.Index(native_idx)
+    if limit is not None:
+        with SqlCounter(query_count=1, join_count=1):
+            method_str = {"bfill": "backfill", "ffill": "pad"}.get(method, method)
+            assert_reindex_exceptions_match(
+                snow_idx,
+                native_idx,
+                lambda idx: idx.reindex(list("CEBFGA"), method=method, limit=limit),
+                ValueError,
+                f"limit argument for '{method_str}' method only well-defined if index and target are monotonic",
+            )
+    else:
+        with SqlCounter(query_count=2, join_count=2):
+            native_result = native_idx.reindex(
+                list("CADEFG"), method=method, limit=limit
+            )
+            snow_result = snow_idx.reindex(list("CADEFG"), method=method, limit=limit)
+            assert_reindex_result_equal(snow_result, native_result)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+def test_reindex_index_ordered_index_unordered_new_index():
```

Comment:
```suggestion
def test_sorted_index_reindexed_to_sorted_new_index():
```

Reply:
This name is wrong, since the new index is unsorted.

Reply:
That's correct, so can we do this?

```suggestion
def test_unsorted_index_reindexed_to_sorted_new_index():
```

I couldn't parse the current function name, and I think you mean to talk about sortedness, not orderedness, because `Index` is always ordered.